src/snowflake/snowpark/mock/_stage_registry.py

```diff
@@ -0,0 +1,232 @@
+    return stage_name, dir_path
+
+
+class StageEntity:
+    def __init__(self, root_dir_path: str, stage_name: str) -> None:
+        self._stage_name = stage_name
+        # stage name might contain special chars which can not be used as dir name
+        # so we generate uuid as name
+        self._dir_name = str(uuid.uuid4())
+        self._working_directory = os.path.join(root_dir_path, self._dir_name)
+
+        if os.path.exists(self._working_directory):
+            shutil.rmtree(self._working_directory)
+
+        os.mkdir(self._working_directory)
+        self._files = set()
+
+    def put_file(
+        self, local_file_name: str, stage_prefix: str, overwrite: bool = False
+    ) -> TableEmulator:
+        local_file_name = local_file_name[
+            len("`file://") : -1
+        ]  # skip normalized prefix `file:// and suffix `
+        # glob supports wildcard '?' and '*' searching
+        list_of_files = glob.glob(local_file_name)
+        result_df = TableEmulator(
+            columns=PUT_RESULT_KEYS,
+            sf_types={
+                "source": ColumnType(StringType(), True),
+                "target": ColumnType(StringType(), True),
+                "source_size": ColumnType(DecimalType(10, 0), True),
+                "target_size": ColumnType(DecimalType(10, 0), True),
+                "source_compression": ColumnType(StringType(), True),
+                "target_compression": ColumnType(StringType(), True),
+                "status": ColumnType(StringType(), True),
+                "message": ColumnType(StringType(), True),
+            },
+            dtype=object,
+        )
+
+        if not list_of_files:
+            raise SnowparkSQLException(f"File doesn't exist: {local_file_name}")
+
+        for local_file_name in list_of_files:
+
+            file_name = os.path.basename(local_file_name)
+            stage_target_dir_path = os.path.join(self._working_directory, stage_prefix)
+
+            if not os.path.exists(stage_target_dir_path):
+                os.makedirs(stage_target_dir_path)
+
+            if (
+                os.path.isfile(os.path.join(stage_target_dir_path, file_name))
+                and not overwrite
+            ):
+                status = "SKIPPED"
+            else:
+                shutil.copy(local_file_name, stage_target_dir_path)
+                status = "UPLOADED"
+
+            file_size = os.path.getsize(os.path.expanduser(local_file_name))
+            result_df.loc[len(result_df)] = {
+                k: v
+                for k, v in zip(
+                    PUT_RESULT_KEYS,
+                    [
+                        file_name,
+                        file_name,
+                        file_size,
+                        file_size,
+                        "NONE",
+                        "NONE",
+                        status,
+                        "",
+                    ],
+                )
+            }
```

Comment:
You can create a dict from a list of pairs instead of doing the whole comprehension.
```suggestion
            values = [file_name, file_name, file_size, file_size, "NONE", "NONE", status, "", ]
            result_df.loc[len(result_df)] = dict(zip(PUT_RESULT_KEYS, values))
```

Reply:
updated